### ISSUE TITLE
Identify returns "Internal Server Error" if no objects of a configured type are found

### ISSUE THREAD

**Issue by RKrahl** [icatproject/icat.oaipmh#3]
If icat.oaipmh does not find at least one object for one of the configured `mainObject` types, the `Identify` call returns HTTP Status 500 - Internal Server Error.

I used the code directly from the github repository at commit 3bc1b8d.  I found this with two different configurations:

At first I used more or less the example `run.properties` file with:
```
data.configurations = inv stud

data.inv.metadataPrefixes = oai_dc oai_datacite
data.stud.metadataPrefixes = oai_dc

data.inv.mainObject = Investigation
data.stud.mainObject = Study
```
The server had one Investigation visible to icat.oaipmh, but no Study. The `Identify` call returned Internal Server Error.

When I disabled Study in the `run.properties` everything seemed to work.  Then I removed the last Investigation and the error appeared again.

I found no trace of the error in the `server.log`.  The `oaipmh.log` had only two lines:
```
2019-07-25 10:44:44,873 INFO [http-thread-pool::jk-connector(1)] RequestInterface - Initialised RequestInterface
2019-07-25 11:01:16,616 ERROR [http-thread-pool::jk-connector(5)] ResponseBuilder - Index: 0
```

**Comment by EmilJunker** [comment]
Thanks! This should now be fixed in a188eba24b6d924d070f3f4f752529bff8c8cd27

**Comment by RKrahl** [comment]
Looks good now: `Identify` works as expected for an "empty" repository, `ListIdentifiers` and `ListRecords` yield a `noRecordsMatch` error code, which is expected as well.

There is another issue with `ListMetadataFormats` in this case, but I'll open an other issue for that.
